src/app/login-button/login-button.component.spec.ts

```diff
@@ -1,23 +1,56 @@
 import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {AuthService} from '../auth.service';
+import {AuthServiceStub} from '../testing/auth-service-stub';
 
 import {LoginButtonComponent} from './login-button.component';
 
 describe('LoginButtonComponent', () => {
+  let authServiceStub: AuthServiceStub;
+  let compiled: HTMLElement;
   let component: LoginButtonComponent;
   let fixture: ComponentFixture<LoginButtonComponent>;
 
+
   beforeEach(async () => {
-    await TestBed.configureTestingModule({declarations: [LoginButtonComponent]})
+    authServiceStub = new AuthServiceStub();
+
+    await TestBed
+        .configureTestingModule({
+          declarations: [LoginButtonComponent],
+          providers: [{provide: AuthService, useValue: authServiceStub}],
+        })
         .compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(LoginButtonComponent);
     component = fixture.componentInstance;
+    compiled = fixture.nativeElement;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
```

Comment:
It looks like the logInOrOut functionality is hooked up in this commit - could you add test cases with mocked out authService.logInOrOut() to show that clicking the button  registers that function?

Reply:
Added test for seeing if the service's `logInOrOut()` function was called. Changes are in the newest commit.